generator/internal/api/model.go

```diff
@@ -144,6 +276,47 @@ type PathInfo struct {
 	//
 	// If this is empty then the body is not used.
 	BodyFieldPath string
+
+	// The method that this path info is associated with.
+	Method *Method
+}
+
+func (p *PathInfo) Accept(v Visitor) error {
+	err := v.VisitPathInfo(p)
+	if err != nil {
+		return err
+	}
+	for _, s := range p.PathTemplate {
+		s.Parent = p
+		err = s.Accept(v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (i InitializationVisitor) VisitPathInfo(p *PathInfo) error {
+	for _, segment := range p.PathTemplate {
+		if segment.FieldPath != nil {
+			// Every FieldPath starts pointing to the root input type Message
```

Comment:
Consider

```suggestion
		if segment.FieldPath == nil {
		    continue
		}
	        // Every FieldPath starts pointing to the root input type Message
```

To save ourselves som indentation.